roachtest: allow large max range size

andrewbaptist opened the pull request:
After the introduction of #129450, there is a default max range size of 4GiB. This test explicitly attempts to set the ranges to 10GiB and fails during setup. This commit increases the max range size for this test.

Fixes: #129556
Fixes: #129552
Fixes: #129551
Fixes: #129544
Fixes: #129542
Fixes: #129540
Fixes: #129541
Fixes: #129539

Epic: none

Release note: None

Commit:
roachtest: allow large max range size

After the introduction of #129450, there is a default max range size of
8GiB. This test explicitly attempts to set the ranges to 10GiB and fails
during setup. This commit increases the max range size for this test.

Fixes: #129556
Fixes: #129552
Fixes: #129551
Fixes: #129544
Fixes: #129542
Fixes: #129540
Fixes: #129541
Fixes: #129539

Epic: none

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/129584)
<!-- Reviewable:end -->

Commit:
kvserver: fix the debugging message for large range

Previously the debug message was printing the min size rather than the
max size.

Epic: none

Release note: None

arulajmani reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 2 of 2 files at r1, 2 of 2 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/129584#-:-O504eV2AYZRk9luXbj3:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @andrewbaptist)

___
*[`-- commits` line 5 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/129584#-O502vHd0-J1PRAZTY1F):*
Is it 4 GiB or 8?
___
*[`pkg/cmd/roachtest/tests/split.go` line 436 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/129584#-O5030gT9gw2VAxR8fZd:-O5030gT9gw2VAxR8fZe:bc87luo) ([raw file](https://github.com/cockroachdb/cockroach/blob/5843688689491322fff9ce218b1393c7f4932c6f/pkg/cmd/roachtest/tests/split.go#L436)):*
> ```Go
> 	)
> 	settings := install.MakeClusterSettings()
> 	if params.maxSize > 8<<30 {
> ```

nit: consider adding a comment.
___
*[`pkg/cmd/roachtest/tests/split.go` line 478 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/129584#-O50357w6zBMfUQG3ISz:-O50357w6zBMfUQG3IT-:bjl1w8z) ([raw file](https://github.com/cockroachdb/cockroach/blob/5843688689491322fff9ce218b1393c7f4932c6f/pkg/cmd/roachtest/tests/split.go#L478)):*
> ```Go
> 		if params.maxSize > 8<<30 {
> 			t.Status("allowing ranges up to ", params.maxSize, " bytes")
> 			_, err := db.ExecContext(ctx, fmt.Sprintf("SET CLUSTER SETTING kv.range.range_size_hard_cap = '%d'", params.maxSize))
> ```

Instead of setting it to params.maxSize, let's set it to 2x that, which is the default backpressure multiplier.
___
*[`pkg/cmd/roachtest/tests/split.go` line 598 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/129584#-O504EXp356jFQkgIKMQ:-O504EXp356jFQkgIKMR:b6x2wxi) ([raw file](https://github.com/cockroachdb/cockroach/blob/5843688689491322fff9ce218b1393c7f4932c6f/pkg/cmd/roachtest/tests/split.go#L598)):*
> ```Go
> 	numNodes := c.Spec().NodeCount
> 	settings := install.MakeClusterSettings()
> 	settings.Env = append(settings.Env, fmt.Sprintf("COCKROACH_MAX_RANGE_MAX_BYTES=%d", 10*size))
> ```

Why 10x? This env var is just checked when applying the zone config, so shouldn't setting it to just size do the trick?
___
*[`pkg/cmd/roachtest/tests/split.go` line 638 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/129584#-O503e2M55SYvMBv_gnd:-O503e2M55SYvMBv_gne:biopqgg) ([raw file](https://github.com/cockroachdb/cockroach/blob/5843688689491322fff9ce218b1393c7f4932c6f/pkg/cmd/roachtest/tests/split.go#L638)):*
> ```Go
> 			}
> 
> 			if _, err := db.ExecContext(ctx, fmt.Sprintf("SET CLUSTER SETTING kv.range.range_size_hard_cap = '%d'", 10*size)); err != nil {
> ```

nit: let's add a comment saying we're effectively disabling the hard cap.
___
*[`pkg/config/zonepb/zone.go` line 413 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/129584#-O504Yhf8h5YYdnxTkn1:-O504Yhf8h5YYdnxTkn2:b-7z474x) ([raw file](https://github.com/cockroachdb/cockroach/blob/5843688689491322fff9ce218b1393c7f4932c6f/pkg/config/zonepb/zone.go#L413)):*
> ```Go
> 	if z.RangeMaxBytes != nil && *z.RangeMaxBytes > maxRangeMaxBytes {
> 		return fmt.Errorf("RangeMaxBytes %d greater than maximum allowed %d",
> 			*z.RangeMaxBytes, maxRangeMaxBytes,
> ```

Thanks :) 


<!-- Sent from Reviewable.io -->

andrewbaptist reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/129584)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @arulajmani)

___
*[`-- commits` line 5 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/129584#-O502vHd0-J1PRAZTY1F):*
<details><summary><i>Previously, arulajmani (Arul Ajmani) wrote…</i></summary><blockquote>

Is it 4 GiB or 8?
</blockquote></details>

Fixed - it was 8 GiB
___
*[`pkg/cmd/roachtest/tests/split.go` line 436 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/129584#-O5030gT9gw2VAxR8fZd:-O505bPDEuFxKNVdQZS3:bb6un1j) ([raw file](https://github.com/cockroachdb/cockroach/blob/5843688689491322fff9ce218b1393c7f4932c6f/pkg/cmd/roachtest/tests/split.go#L436)):*
<details><summary><i>Previously, arulajmani (Arul Ajmani) wrote…</i></summary><blockquote>

nit: consider adding a comment.
</blockquote></details>

Done
___
*[`pkg/cmd/roachtest/tests/split.go` line 478 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/129584#-O50357w6zBMfUQG3ISz:-O505pZ51Mk2GocVH4NZ:bb6un1j) ([raw file](https://github.com/cockroachdb/cockroach/blob/5843688689491322fff9ce218b1393c7f4932c6f/pkg/cmd/roachtest/tests/split.go#L478)):*
<details><summary><i>Previously, arulajmani (Arul Ajmani) wrote…</i></summary><blockquote>

Instead of setting it to params.maxSize, let's set it to 2x that, which is the default backpressure multiplier.
</blockquote></details>

Done
___
*[`pkg/cmd/roachtest/tests/split.go` line 598 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/129584#-O504EXp356jFQkgIKMQ:-O506QKd0Mo6yCK7l0OP:b-i7fmus) ([raw file](https://github.com/cockroachdb/cockroach/blob/5843688689491322fff9ce218b1393c7f4932c6f/pkg/cmd/roachtest/tests/split.go#L598)):*
<details><summary><i>Previously, arulajmani (Arul Ajmani) wrote…</i></summary><blockquote>

Why 10x? This env var is just checked when applying the zone config, so shouldn't setting it to just size do the trick?
</blockquote></details>

The test was multiplying by 10 below. I pulled this up into a variable to make this more clear.
___
*[`pkg/cmd/roachtest/tests/split.go` line 638 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/129584#-O503e2M55SYvMBv_gnd:-O506bviBrE6E54u0K-C:bb6un1j) ([raw file](https://github.com/cockroachdb/cockroach/blob/5843688689491322fff9ce218b1393c7f4932c6f/pkg/cmd/roachtest/tests/split.go#L638)):*
<details><summary><i>Previously, arulajmani (Arul Ajmani) wrote…</i></summary><blockquote>

nit: let's add a comment saying we're effectively disabling the hard cap.
</blockquote></details>

Done
___
*[`pkg/config/zonepb/zone.go` line 413 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/129584#-O504Yhf8h5YYdnxTkn1:-O506d6411ktylqplsOW:b87tees) ([raw file](https://github.com/cockroachdb/cockroach/blob/5843688689491322fff9ce218b1393c7f4932c6f/pkg/config/zonepb/zone.go#L413)):*
<details><summary><i>Previously, arulajmani (Arul Ajmani) wrote…</i></summary><blockquote>

Thanks :)
</blockquote></details>

NP! It took me a minute to understand why the size was wrong.


<!-- Sent from Reviewable.io -->

andrewbaptist commented:
TFTR!

bors r=arulajmani
